public/src/components/calls/callcntrl_btn_group.jsx

```diff
@@ -1,17 +1,18 @@
 import React from 'react';
-import { Button } from 'react-bootstrap';
+import { Button, HelpBlock } from 'react-bootstrap';
 
 const CallControl = (props) => {
-  const { handler, text, outcome, status, htmlID } = props;
+  const { handler, text, outcome, status, htmlID, invalid } = props;
   const style = text === 'Hang Up' ? 'danger' : 'success';
-  const disableStatus = text === 'Hang Up' ? (outcome === 'PENDING' || status !== 'IN_PROGRESS') : (outcome === 'PENDING' && status !== 'HUNG_UP');
+  const disableStatus = (text === 'Hang Up') ? (outcome === 'PENDING' || status !== 'IN_PROGRESS') : ((outcome === 'PENDING' && status !== 'HUNG_UP') || (outcome === 'ANSWERED' && status !== 'HUNG_UP') || invalid);
```

Comment:
Looks like we smoshed together a lot of button logic in here. This may surface a few bugs later, but let's fix them later as opposed to pulling this apart, which will take a while. Ideally each button's disabled logic should live as close to the button as possible, making it easier to validate it's logic.

Reply:
I agree, it took me a while to figure to even read this out myself. Should there be a ticket for it @perfectlynormalbeast ?

Reply:
It would be great to add a ticket to the backlog pointing to this ratsnest and how to change things up.